src/query_parser/v2/parser.y

```diff
@@ -1195,21 +1195,23 @@ term(A) ::= TERM(B) . {
 
 term(A) ::= NUMBER(B) . {
   A = B;
+  A.type = QT_NUMERIC;
 }
 
 term(A) ::= SIZE(B). {
   A = B;
+  A.type = QT_SIZE;
 }
 
 ///////////////////////////////////////////////////////////////////////////////////
 // Parameterized Primitives (actual numeric or string, or a parameter/placeholder)
 ///////////////////////////////////////////////////////////////////////////////////
 
-
-// Number is treated as a term here
 param_term(A) ::= term(B). {
   A = B;
-  A.type = QT_TERM;
+  if((A.type != QT_NUMERIC) && (A.type != QT_SIZE)) {
+    A.type = QT_TERM;
```

Comment:
What would it have been before (the type)?
And what will it be for these two cases?

Reply:
Before the change, all the terms were handle as `QT_TERMS`, inclusive the numbers. See this comment in master:
https://github.com/RediSearch/RediSearch/blob/bda189a9dfca9f51c88284eb3939796710f28315/src/query_parser/v2/parser.y#L1209

Now, we need to preserve the correct type for `NUMBER` (`QT_NUMERIC`) and `SIZE` (`QT_SIZE`) to set the flag `QueryNode_Verbatim`.

Reply:
Then why do we need this condition and assignment at all?
If the type is set by the rules and we now wish to keep it, isn't this redundant?

Reply:
OK. I removed the condition and assignment, now the type is set by the rules. For the `term` I was not setting the type but now it was fixed.